File: core/src/engine/rts/model/OptionBase.scala
package engine.rts.model

import javax.swing.JSlider

class OptionBase (val numSlider:JSlider){

    val defaultMapDir = "mapData/TinyMap/"
    private var mapName=defaultMapDir//This is the name of the map we will use
    private var time=0.0
    val maxTime=Math.pow(10,9)*15*60 //This is used to represent how much time is left in the game.
    /**
     * These functions are used to keep track of how much time has elapsed in the game.
     */
    def startTime(){
        time=System.nanoTime()
    }

    def getTimeElapsed()= System.nanoTime()-time
    def getRemainingTime()=maxTime-getTimeElapsed()
    def isDone()=getTimeElapsed()<maxTime
    def getPlayerCount()=numSlider.getValue()
    def setMapName(name:String){
        mapName=name.substring(0, name.length()-4)
    }
    def getMapName()=mapName
}

File: core/src/engine/rts/model/OptionBase.java
package engine.rts.model;

import javax.swing.JSlider;

public abstract class OptionBase{
	
    final static String DEFALUT_MAP = "mapData/TinyMap/";    		
	final int numSlider;

    private String mapName=DEFALUT_MAP;//This is the name of the map we will use
    private double time=0.0;

    //TODO:Make sure this value is not 
    final double maxTime=(int)Math.pow(10,9)*15*60; //This is used to represent how much time is left in the game.

	public OptionBase(int n){
		numSlider=n;
	}

    /**
     * These functions are used to keep track of how much time has elapsed in the game.
     */
    public void startTime(){
        time=System.nanoTime();
    }

    public double getTimeElapsed(){
    	return System.nanoTime()-time;
    }
    
    public double getRemainingTime(){
    	return maxTime-getTimeElapsed();
    }
    
    public boolean isDone(){
    	return getTimeElapsed()<maxTime;
    }
    
    public int getPlayerCount(){
    	return numSlider;
    }
    
    public void setMapName(String name){
        mapName=name.substring(0, name.length()-4);
    }
    
    public String getMapName(){
    	return mapName;
    }
}
